app/views.py

```diff
@@ -8,3 +8,29 @@
 @App.route('/')
 def public_timeline():
     return render_template('public_timeline.html', flashes=db_utils.get_all_twittes())
+
+@app.route('/register', methods=['GET', 'POST']) # akef
+def register():
+    """Registers the user."""
+    if g.user:
+        return redirect(url_for('timeline'))
+    error = None
+    if request.method == 'POST':
+        if not request.form['username']:
+            error = 'You have to enter a username'
+        elif not request.form['email'] or \
+                '@' not in request.form['email']:
+            error = 'You have to enter a valid email address'
+        elif not request.form['password']:
+            error = 'You have to enter a password'
+        elif request.form['password'] != request.form['password2']:
+            error = 'The two passwords do not match'
+        elif get_user_id(request.form['username']) is not None:
+            error = 'The username is already taken'
+        else:
+            db = get_db()
+            register_user(request.form['username'],request.form['email'],request.form['password'])
```

Comment:
انا قرأت الكود بتاع db_execute ومش شايف اللوجيك ده فيه خااالص
وتمام بالنسبة للرسالة الأخيرة هبقي اعدلها

On Sun, Aug 14, 2016 at 12:11 AM, Ahmed Maher <notifications@github.com>
wrote:

> In app/views.py
> <https://github.com/CATWebTeam/git-workshop/pull/5#discussion_r74692851>:
>
> > +    error = None
> > +    if request.method == 'POST':
> > +        if not request.form['username']:
> > +            error = 'You have to enter a username'
> > +        elif not request.form['email'] or \
> > +                '@' not in request.form['email']:
> > +            error = 'You have to enter a valid email address'
> > +        elif not request.form['password']:
> > +            error = 'You have to enter a password'
> > +        elif request.form['password'] != request.form['password2']:
> > +            error = 'The two passwords do not match'
> > +        elif get_user_id(request.form['username']) is not None:
> > +            error = 'The username is already taken'
> > +        else:
> > +            db = get_db()
> > +            register_user(request.form['username'],request.form['email'],request.form['password'])
>
> this will give you an register_user not defined ERROR
> you should use db_utils.register_user insted
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/CATWebTeam/git-workshop/pull/5/files/91463a8b79ad1996ef7c7f3c0e41010ccbf6565a#r74692851>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMhI18eI-IxyTu1Zd8To6REhixThs4g6ks5qfkEvgaJpZM4Jjw-i>
> .
>